File: app/src/main/java/com/business/toos/basedialog/utils/ToastDialog.java
package com.business.toos.basedialog.utils;

import android.view.View;

import androidx.appcompat.widget.AppCompatImageView;
import androidx.appcompat.widget.AppCompatTextView;
import com.business.toos.R;


/**
 * Created by Administrator on 2019/10/9.
 */

public class ToastDialog extends BaseFragDialog {

    private Type mType;
    private String mMessage;
    private AppCompatImageView icon;
    private AppCompatTextView message;
    private long mUptimeMillis;
    private ProgressView progressView;

    public ToastDialog(Object view, float alpha, boolean autoDismiss, boolean cancelable, int animation, int gravity) {
        super(view, alpha, autoDismiss, cancelable, animation, gravity);
    }

    public static DialogBuilder<ToastDialog> ToastBuilder() {
        return new DialogBuilder<>(ToastDialog.class);
    }

    @Override
    public void initView(View view) {
        icon = view.findViewById(R.id.iv_toast_icon);
        message = view.findViewById(R.id.tv_toast_message);
        progressView = view.findViewById(R.id.pw_progress);
        if (mType == null) {
            throw new NullPointerException("没有设置类型，请调用 setType() 设置类型");
        }
        crate();
        //取消背景遮罩
        setBackgroundDimEnabled(false);
    }

    private void crate() {
        icon.setVisibility(View.VISIBLE);
        progressView.setVisibility(View.GONE);
        switch (mType) {
            case FINISH:
                icon.setImageResource(R.drawable.ic_dialog_finish);
                break;
            case ERROR:
                icon.setImageResource(R.drawable.ic_dialog_error);
                break;
            case WARN:
                icon.setImageResource(R.drawable.ic_dialog_warning);
                break;
            case LOADING:
                icon.setVisibility(View.GONE);
                progressView.setVisibility(View.VISIBLE);
                break;
            default:
                break;
        }
        if (mMessage == null) {
            message.setVisibility(View.GONE);
        } else {
            message.setVisibility(View.VISIBLE);
            message.setText(mMessage);
        }
        if (mUptimeMillis > 0) {
            postAtTime(mUptimeMillis, new Runnable() {
                @Override
                public void run() {
                    dismiss();
                }
            });
        }
    }

    /**
     * 设置显示类型
     *
     * @param type 类型
     */
    public ToastDialog setType(Type type) {
        this.mType = type;
        return this;
    }

    /**
     * 设置消息内容
     */
    public ToastDialog setMessage(String message) {
        this.mMessage = message;
        return this;
    }

    /**
     * 延时关闭
     *
     * @param uptimeMillis 时间，毫秒为单位
     */
    public ToastDialog postAtTime(long uptimeMillis) {
        this.mUptimeMillis = uptimeMillis;
        return this;
    }


    /**
     * 显示的类型
     */
    public enum Type {
        // 完成，错误，警告,  加载中
        FINISH, ERROR, WARN, LOADING
    }
}

File: bullet_ui/src/main/java/com/example/ui/dialog/ToastDialog.kt
package com.example.ui.dialog

import android.content.Context
import android.view.View
import androidx.appcompat.widget.AppCompatImageView
import com.example.ui.R
import com.example.ui.customView.ProgressView
import com.example.ui.dialog.base.FastDialog

/**
 * @name ToastDialog
 * @package com.example.ui.dialog
 * @author 345 QQ:1831712732
 * @time 2020/5/18 21:54
 * @description 通用的 Toast 对话框
 */

object ToastDialog {


    private var fastDialog: FastDialog? = null

    private fun createDialog(context: Context) {
        fastDialog = FastDialog.Builder(context)
                .setContentView(R.layout.dialog_toast)
                .setAlpha(1f)
                .build()
        fastDialog?.setOnDismissListener {
            fastDialog = null
        }
    }

    fun loading(context: Context) {
        createDialog(context)
        fastDialog?.getView<AppCompatImageView>(R.id.iv_toast_icon)?.visibility = View.GONE
        fastDialog?.getView<ProgressView>(R.id.pw_progress)?.visibility = View.VISIBLE
        fastDialog?.setText(R.id.tv_toast_message, "加载中")
        fastDialog?.show()
    }

    fun error(context: Context) {
        createDialog(context)
        show("错误", R.drawable.ic_dialog_error)
    }

    fun warn(context: Context) {
        createDialog(context)
        show("警告", R.drawable.ic_dialog_warning)
    }

    fun finish(context: Context) {
        createDialog(context)
        show("完成", R.drawable.ic_dialog_finish)
    }

    fun stop() {
        if (fastDialog != null) {
            fastDialog?.dismiss()
            fastDialog = null
        }
    }

    fun unLoading() {
        fastDialog?.dismiss()
    }

    fun show(context: Context, text: String) {
        createDialog(context)
        fastDialog?.getView<ProgressView>(R.id.pw_progress)?.visibility = View.GONE
        fastDialog?.getView<AppCompatImageView>(R.id.iv_toast_icon)?.visibility = View.GONE
        fastDialog?.setText(R.id.tv_toast_message, text)
        fastDialog?.show()
    }


    private fun show(text: String, imageResId: Int) {
        hideLoad()
        fastDialog?.getView<AppCompatImageView>(R.id.iv_toast_icon)
                ?.setBackgroundResource(imageResId)
        fastDialog?.setText(R.id.tv_toast_message, text)
        fastDialog?.show()
    }


    private fun hideLoad() {
        fastDialog?.getView<ProgressView>(R.id.pw_progress)?.visibility = View.GONE
        fastDialog?.getView<AppCompatImageView>(R.id.iv_toast_icon)?.visibility = View.VISIBLE
    }

}